### PR TITLE
Conditionally use module definition for typescript types

### PR DESCRIPTION
**Pull request by zOadT**
This PR tries to solve the "conflict" between the use of `namespace` and `module` in the typescript definitions. In a module environment the types should work just as they are written as module (maybe you get a global `planck` as autocomplete-suggestion, but TS will complain if you try to use it).
Otherwise they should act as a namespace just as before this commit.

In module environments this may be a breaking change, but setting `--allowumdglobalaccess` to false should fix it.

More information can be found [here](https://devblogs.microsoft.com/typescript/announcing-typescript-3-5/#the---allowumdglobalaccess-flag).

### CODE REVIEW

**Comment by zOadT** [comment]
Found an issue: `export interface Body {` and `export let Body {` (the same with `world` etc.) seem to override each other 🙁 Currently the only solution I can think of is renaming the interfaces, but that doesn't sound like a good idea...

**Comment by zOadT** [comment]
It seems like the only problem was that I have to put `planck` in front of some types, so TS can resolve them. Right now it looks like it works, but I will wait for some further testing.

File: lib/index.d.ts [hunk 1]
```diff
@@ -2,474 +2,596 @@
...
+  fixedRotation: boolean,
+  bullet: boolean,
+  gravityScale: number,
+  allowSleep: boolean,
+  awake: boolean,
+  active: boolean,
+  userData: any,
+}>;
+
+export class Body {
```
Comment (shakiba) [inline lib/index.d.ts]: do we need both this and `export interface Body`? it seems this class is a copy of the interface.

File: lib/index.d.ts [hunk 2]
```diff
@@ -2,474 +2,596 @@
...
-    new(def: WorldDef): World;
-    (def: WorldDef): World;
-
-    new(gravity: Vec2): World;
-    (gravity: Vec2): World;
-
-    new(): World;
-    (): World;
-  };
+export as namespace planck;
```
Comment (shakiba) [inline lib/index.d.ts]: what would happen if we don't export a namespace?

File: lib/index.d.ts (hunk 1 above)
Comment (zOadT) [reply lib/index.d.ts]: Good catch! I was experimenting with using classes and committed it by accident.

File: lib/index.d.ts (hunk 2 above)
Comment (zOadT) [reply lib/index.d.ts]: You use the namespace when you are in a non-module environment. That means you just write `import "planck-js";` on top of a file or add `"planck-js"` to the `types` field in the tsconfig file to make TS aware of the typings. In this case TS assumes there is a global `planck` object.
So if you don't export the namespace you can only use the typings in a module environment.

**Pull request merged**